wikimedia/wikipedia-ios
# Update AaaLD modal timestamps to be relative and absolute as necessary

**Phabricator:** https://phabricator.wikimedia.org/T264802

### Notes
Updates the display timestamps in the article as a living document modal.

- **Section Header Titles**: Relative to current day (e.g. Today, Yesterday, X days ago, X months ago)
- **Section Header Subtitles**: Absolute dates or date ranges mixing relative and absolute (e.g. November 5, 2020, November 1 - November 5th, 2020, Yesterday - November 1st, 2020)
- **Event timestamps**: Relative by minute if under an hour, by hour if within 24 hours, then GMT (10 minutes ago, 12 hours, 02:40 GMT)

As we are dealing with dates, I just can't be convinced everything is 100%. And as I mentioned to Toni, I think working on this has exposed some other potential quirks with bucketing the sections (depending on the article). But in general regarding the timestamps, the delicate incantations in this PR are seemingly turning up expected results. 

### Test Steps
1. Open an article, then the AaaLD modal.
2. Observe the timestamps adhere to the formats specified above.

Articles I used often during testing:
https://en.wikipedia.org/wiki/A_History_of_the_Birds_of_Europe
https://en.wikipedia.org/wiki/Ornithology

### Screenshot
<img width="500" alt="timestamps" src="https://user-images.githubusercontent.com/5652327/98315878-20245080-1f8e-11eb-9b2c-b44b74426f2f.png">
